compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1189,36 +1189,30 @@ private BField createFieldWithType(BField field, List<BType> bTypes) {
     }
 
     /**
-     * Get the resultant union or non-union type from a {@code List<BType>}.
+     * Get the resultant type from a {@code List<BType>}.
      *
      * @param bTypes bType list (size > 0)
      * @return {@code BUnionType} if effective members in list is > 1. {@code BType} Otherwise.
      */
-    private BType getResultantUnionOrNonUnionType(List<BType> bTypes) {
-        if (bTypes.size() == 1) {
-            return bTypes.get(0);
-        }
```

Comment:
We had missed getting the reference type for size == 1 scenario. A test case was also added for this in this pr.